packages/ui/src/modal/buy/BuyModalRenderer.tsx

```diff
@@ -320,9 +323,11 @@ export const BuyModalRenderer: FC<Props> = ({
   const { data: balance } = useBalance({
     address: address,
     token:
-      currency?.symbol !== 'ETH'
+      currency?.contract !== constants.AddressZero
         ? (currency?.contract as UseBalanceToken)
         : undefined,
+    chainId:
```

Comment:
shouldn't we be passing in chain id no matter what?

Reply:
It's a good point, let me try adding that to see how it works.

Reply:
Done.